src/FSharpPlus/Control/Functor.fs

```diff
@@ -112,6 +113,7 @@ type Map with
     static member        Map ((x: IReadOnlyDictionary<_,_>, f: 'T->'U), _mthd: Default2) = IReadOnlyDictionary.map f x : IReadOnlyDictionary<'Key,_>
     static member        Map ((x: IObservable<'T>         , f: 'T->'U), _mthd: Default2) = Observable.map f x       : IObservable<'U>
     static member        Map ((x: Nullable<_>             , f: 'T->'U), _mthd: Default2) = Nullable.map f x         : Nullable<'U>
+    static member        Map ((x: IReadOnlyCollection<'T> , f: 'T->'U), _mthd: Default1) = IReadOnlyCollection.map f x : IReadOnlyCollection<'U>
```

Comment:
Adding an overload here should probably be tested

Reply:
Perhaps such a simple test as just a:
```f#
let readonlyC = ReadOnlyCollection.ofList [1;2;3]
let mapped = map ((+) 1) readonlyC
Assert.IsInstanceOf< IReadOnlyCollection <int>>(mapped)
```
?

Reply:
See the other `IsInstance` tests we have. Normally we have to wrap them in an `option` to avoid biasing type inference which would defeat the whole test purpose.

Reply:
I've added this test according to your snippet (including gusty's suggestion on option)